homework/index.js

```diff
@@ -30,17 +15,114 @@
+    createAndAppend('td', tr2, { text: repository.description, class: 'value' });
+
+    // raw 3
+    const tr3 = createAndAppend('tr', repInfoTable);
+    createAndAppend('td', tr3, { class: 'heading', text: 'Forks:' });
+    createAndAppend('td', tr3, { text: repository.forks, class: 'value' });
+
+    // raw 4
+    const tr4 = createAndAppend('tr', repInfoTable);
+    createAndAppend('td', tr4, { class: 'heading', text: 'Updated:' });
+    createAndAppend('td', tr4, { text: repository.updated_at, class: 'value' });
+  }
+
+  // function renderError to avoid DRY.
+  function renderError(error, parent) {
+    createAndAppend('div', parent, { text: error.message, class: 'alert-error' });
   }
 
+  // Function: display the contributors of the selected repository ( right side ).
+  async function showContributors(contributorURL, container) {
+    createAndAppend('h2', container, { text: 'Contributions' });
+    const ul = createAndAppend('ul', container);
+    try {
+      const fetched = await fetch(contributorURL);
+      const contributorsData = await fetched.json();
+      contributorsData.forEach(contributor => {
+        const li = createAndAppend('li', ul);
+        const a = createAndAppend('a', li, {
+          href: contributor.html_url,
+          target: '_blank',
+          text: contributor.login,
+        });
+        createAndAppend('img', a, {
+          src: contributor.avatar_url,
+          width: 100,
+          class: 'spacing',
+        });
+      });
+    } catch (error) {
+      renderError(error, container);
+    }
+  }
+
+  // function main:
+  async function main(url) {
+    const root = document.getElementById('root');
+
+    // the main title.
+    createAndAppend('h1', root, { text: 'HYF Repositories' });
+
+    try {
+      // fetch (ALL DATA).
+      const fetchedUrl = await fetch(url);
+      const data = await fetchedUrl.json();
+
+      const theContainer = createAndAppend('div', root, { id: 'container' });
+
+      // the basic repository information (left aside).
+      const basicRepoInfoAside = createAndAppend('aside', theContainer, { class: 'box' });
+
+      // select element.
+      const select = createAndAppend('select', theContainer);
+
+      // the repository contributors (right aside).
+      const contributorsAside = createAndAppend('aside', theContainer, { class: 'box' });
+
+      // sorting the repositories (case-insensitive) on repository name.
+      data.sort((a, b) => a.name.localeCompare(b.name));
+
+      data.forEach((repo, index) => {
+        // appending repositories as options to the select element.
+        basicRepoInfoAside.innerHTML = '';
+        contributorsAside.innerHTML = '';
+        createAndAppend('option', select, { text: repo.name, value: index });
+        showRepoBasicInfo(basicRepoInfoAside, data[0]);
+        showContributors(data[0].contributors_url, contributorsAside);
+
+        // change select:
+        select.addEventListener('change', () => {
```

Comment:
you'r having your `eventListener` inside your `forEach`, which is okay in your case as a very small application, but in a real worl you need to have it seperated, you will know why in a later advanced classes.